File: examples/server/routes/languages/php/yon.php
<?php

require_once getcwd() . '/server/services/php_language_service.php';

class Handler {
    private const RESPONSES = [
        '406' => ['code' => '406', 'detail' => 'not acceptable'],
        '407' => ['code' => '407', 'detail' => 'proxy auth required'],
        '408' => ['code' => '408', 'detail' => 'request timeout'],
        '409' => ['code' => '409', 'detail' => 'conflict'],
        '410' => ['code' => '410', 'detail' => 'gone'],
    ];

    public static function GET($request) {
        $code = self::statusCode($request);
        if (array_key_exists($code, self::RESPONSES)) {
            return self::RESPONSES[$code];
        }

        return self::service()->describe($request);
    }

    private static function statusCode($request): string {
        $query = $request['query'] ?? [];
        $raw = $query['code'] ?? '';
        return is_numeric($raw) ? (string) intval($raw) : (string) $raw;
    }

    private static function service(): PhpLanguageService {
        static $service = null;
        if ($service === null) {
            $service = new PhpLanguageService();
        }
        return $service;
    }
}

File: examples/server/routes/languages/ruby/yon.rb
require_relative '../../../services/ruby_language_service'

class Handler
  STATUS_RESPONSES = {
    '401' => { code: '401', detail: 'unauthorized' },
    '402' => { code: '402', detail: 'payment required' },
    '403' => { code: '403', detail: 'forbidden' },
    '404' => { code: '404', detail: 'not found' },
    '405' => { code: '405', detail: 'method not allowed' }
  }.freeze
  private_constant :STATUS_RESPONSES

  def self.GET(request)
    code = status_code(request)
    return STATUS_RESPONSES[code] if STATUS_RESPONSES.key?(code)

    service.describe(request)
  end

  def self.status_code(request)
    query = request['query'] || {}
    query['code'] ? query['code'].to_s : ''
  end

  def self.service
    @service ||= RubyLanguageService.new
  end

  private_class_method :status_code, :service
end
